diff --git a/src/crypto/randomx/randomx.cpp b/src/crypto/randomx/randomx.cpp
index 0f86ec04..71e6acc5 100644
--- a/src/crypto/randomx/randomx.cpp
+++ b/src/crypto/randomx/randomx.cpp
@@ -56,6 +56,10 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define XMRIG_FORCE_JIT_32B_BRANCH_ALIGNMENT 0
 #endif
 
+#ifndef XMRIG_FORCE_NO_BMI2
+#define XMRIG_FORCE_NO_BMI2 0
+#endif
+
 RandomX_ConfigurationMoneroV2::RandomX_ConfigurationMoneroV2()
 {
 	ProgramSize = 384;
@@ -196,7 +200,7 @@ RandomX_ConfigurationBase::RandomX_ConfigurationBase()
 		const uint8_t* b = addr(randomx_sshash_end);
 		memcpy(codeSshPrefetchTweaked, a, b - a);
 	}
-	if (xmrig::Cpu::info()->hasBMI2()) {
+	if (xmrig::Cpu::info()->hasBMI2() && !XMRIG_FORCE_NO_BMI2) {
 		const uint8_t* a = addr(randomx_prefetch_scratchpad_bmi2);
 		const uint8_t* b = addr(randomx_prefetch_scratchpad_end);
 		memcpy(codePrefetchScratchpadTweaked, a, b - a);
@@ -243,7 +247,7 @@ void RandomX_ConfigurationBase::Apply()
 	//*(uint32_t*)(codeReadDatasetTweaked + 24) = DatasetBaseMask;
 	//*(uint32_t*)(codeReadDatasetLightSshInitTweaked + 59) = DatasetBaseMask;
 
-	const bool hasBMI2 = xmrig::Cpu::info()->hasBMI2();
+	const bool hasBMI2 = xmrig::Cpu::info()->hasBMI2() && !XMRIG_FORCE_NO_BMI2;
 
 	*(uint32_t*)(codePrefetchScratchpadTweaked + (hasBMI2 ? 7 : 4)) = ScratchpadL3Mask64_Calculated;
 	*(uint32_t*)(codePrefetchScratchpadTweaked + (hasBMI2 ? 17 : 18)) = ScratchpadL3Mask64_Calculated;
 
